### PR TITLE
Myprofile

### PR DESCRIPTION
**Pull request by EduardoLopezPalomo**
The user can change everithing except the firstname, the lastname and the password. 
There is a pop up when dor the user to confirm the changes

### CODE REVIEW

File: client/src/components/MyProfile.js
Comment (jannejjj) [inline client/src/components/MyProfile.js:52]: Here you can use /users/updateProfile for the url so the request is proxied to the backend correctly using the backend address from the environment variable

File: client/src/components/MyProfile.js
Comment (jannejjj) [inline client/src/components/MyProfile.js:219]: Same comment about the url, it should work with just /api/getID

File: client/src/components/MyProfile.js
Comment (jannejjj) [inline client/src/components/MyProfile.js:231]: Url

File: client/src/components/MyProfile.js
Comment (jannejjj) [inline client/src/components/MyProfile.js:134]: Personally, I think I would pass the setters themselves to the Modal as props so you wouldn't need to have separate functions in this file🤔 Something like:
```
<EditDetailsModal
      setEditedFirstname={setEditedFirstname}
      setEditedLastname={setEditedLastname}
      .
      .
      />
```
and then in the modal you could have
```
onInput={(event) => {setEditedFirstname(event.target.value)}}
```

But it's also fine like this for me, up to you if you want to change or not :D

File: client/src/components/MyProfile.js
Comment (jannejjj) [inline client/src/components/MyProfile.js:10]: I think these should not have capital first letters, same with the history variables  :P

File: client/src/components/MyProfile.js
Comment (jannejjj) [inline client/src/components/MyProfile.js:223]: I think we discussed that we're gonna use localStorage, to make the data persistent if you close the browser for example

File: client/src/components/MyProfile.js
Comment (jannejjj) [inline client/src/components/MyProfile.js:243]: logging leftovers from developing I guess :D

File: client/src/components/MyProfile.js
Comment (jannejjj) [inline client/src/components/MyProfile.js:47]: This comment is unnecessary, right? That's what we're doing, sending the values, no? 🤪

**Review by jannejjj — CHANGES_REQUESTED** [review]
Sorry for so many comments, not critisizing your work, just some minor things I noticed that should be easy to change!

I think change the urls, variable casing and switch to localStorage instead of sessionStorage before continuing 🤔

**Comment by Nesssu** [comment]
AssocEase_MyProfileModal_InputOrder

Move the email input field below the phone input field. It makes more sense to be there

**Comment by Nesssu** [comment]
AssocEase_MyProfilePage_ElementOrder

Change the element alignment to vertical. Item order would be like this:

- Profile picture
- User information
- Button to edit information

The user information can be made to look better on the page. In my opinion it would be better if it was like this for example:

- Daniel Caesar
- +3582349650
- daniel.caesar@email.com (Add the email that is currently missing)
- Mäkikatu 3
- 53850, Lappeenranta
- Finland

Also it is pointless (on the profile page) to show the title for the users' name, email, phone, etc. I think everyone using this website can tell what their name is without us having to literally show that this is your firstname and this your last. Same thing for the other fields. It's quite easy to know what is the phone number and what is the email address.

File: client/src/components/MyProfile.js
Comment (Nesssu) [reply client/src/components/MyProfile.js:223]: Also we shouldn't store user information to the local/sessionStorage. That is a public storage and other websites could then use that id which is not a good thing. Best way to store the user id is to have an user object that has all the data that we want to know about the current user. I made a PR that creates this object for us. (Token to localStorage and authentication #20) 

If you have any questuions, you can ask me.

File: client/src/components/MyProfile.js
Comment (EduardoLopezPalomo) [reply client/src/components/MyProfile.js:231]: I don't know why, but without the localhost the fetch doesnt work, none of them.

**Comment by EduardoLopezPalomo** [comment]
I think I cahnge everything, but i cuoldnt test it because i change the localstorage thing and I dont have merge #20 branch

File: client/src/components/MyProfile.js
Comment (jannejjj) [reply client/src/components/MyProfile.js]: Have you set the environment variables in client/.env ? It should contain
`REACT_APP_PROXY_TARGET='http://localhost:4000'

😄

**Comment by jannejjj** [comment]
Yeah, I guess for testing the localStorage part of this PR, you can either:
1) wait for Nestori's PR to be merged and then merge main to your branch, or
2) Merge Nestori's branch straight to your branch. 

I recommend waiting for @Nesssu 's PR to be merged, and then merging main to your branch. That's probably the least confusing option 😄 After that, you can see test that everything works fully.

**Comment by Nesssu** [comment]
You can now merge the main branch to this branch and we can start testing

**Review by Nesssu — APPROVED** [review]

File: client/src/components/MyProfile.js
Comment (Jezerig) [reply client/src/components/MyProfile.js]: Has this been fixed?

Comment (EduardoLopezPalomo) [reply client/src/components/MyProfile.js]: I think so

**Comment by jannejjj** [comment]
Edu, my bad with the urls.. The reason why it isn't working without the `localhost` part is that in our setupProxy.js we only define a proxy for the `/api` route, when there should also be one for `/users` ..

So, the urls should work as intended with this change into setupProxy.js:

image

After that, you can finally merge this PR. I think. Sorry again for the confusion!

**Review by jannejjj — APPROVED** [review]

**Review by jannejjj — CHANGES_REQUESTED** [review]
Just make the proxy change, then you're done with this!

**Review by Jezerig — APPROVED** [review]

**Review by jannejjj — APPROVED** [review]

**Pull request merged**
